src/guacd/proc.c

```diff
@@ -84,7 +102,11 @@ static void* guacd_user_thread(void* data) {
     guac_client* client = proc->client;
 
     /* Get guac_socket for user's file descriptor */
+#ifdef CYGWIN_BUILD
+    guac_socket* socket = guac_socket_open_handle(params->handle);
+#else
     guac_socket* socket = guac_socket_open(params->fd);
+#endif
     if (socket == NULL)
```

Comment:
Could this prevent params from being freed?

Reply:
Uh yeah, I guess so. Looks like this was a potential issue before, but none of the `guac_socket*` functions look like they'll ever actually return `NULL` so in practice it wouldn't happen. Better safe than sorry though.